src/plugins/intel_gna/src/backend/gna_limitations.cpp

```diff
@@ -617,16 +673,83 @@ bool Validator_35::ShouldUseOnlyConv2DGnaIface() const {
+    struct CnnLimits {
+        const RangeLimit2D kInputHWLimit;
+        const RangeMultipleLimit kInputChannelsNumberLimit;
+        const RangeLimit2D kKerneHWlLimit;
+        const RangeLimit2D kStrideHWLimit;
+        const RangeLimit2D kDilationLimit;
+        const RangeLimit2D kPoolingWindowHWLimit;
+        const RangeLimit2D kPoolingStrideHWLimit;
+    };
+
+    static const CnnLimits kDwscLimits;
+
+public:
+    Validator_36() = default;
+
+    bool ValidateDwsc(const std::string& name,
+                      const uint32_t inHeight,
+                      const uint32_t inWidth,
+                      const uint32_t inChannels,
+                      const uint32_t kH,
+                      const uint32_t kW,
+                      const uint32_t kN,
+                      const uint32_t strideH,
+                      const uint32_t strideW,
+                      const uint32_t dilationH,
+                      const uint32_t dilationW,
+                      bool exception = true) const override;
+};
+
+const Validator_36::CnnLimits Validator_36::kDwscLimits{
+    {{1, 65535, "input height"}, {1, 65535, "input width"}},                        // kInputHWLimit
+    {{8, 1024, "number of input channels"}, 8},                                     // kInputChannelsNumberLimit
+    {{1, 255, "kernel height"}, {1, 255, "kernel width"}},                          // kKerneHWlLimit
+    {{1, 255, "convolution stride height"}, {1, 255, "convolution stride width"}},  // kStrideHWLimit
+    {{Limitations::kConvDilationHeight, Limitations::kConvDilationHeight, "dilation height"},  // kDilationLimit
+     {Limitations::kConvDilationWidth, Limitations::kConvDilationWidth, "dilation width"}},
+    {{1, 255, "pooling window height"}, {1, 255, "pooling window width"}},  // kPoolingWindowHWLimit
+    {{1, 255, "pooling stride height"}, {1, 255, "pooling stride width"}}   // kPoolingStrideHWLimit
+};
+
+bool Validator_36::ValidateDwsc(const std::string& name,
+                                const uint32_t inHeight,
+                                const uint32_t inWidth,
+                                const uint32_t inChannels,
+                                const uint32_t kernelH,
+                                const uint32_t kernelW,
+                                const uint32_t kernelN,
+                                const uint32_t strideH,
+                                const uint32_t strideW,
+                                const uint32_t dilationH,
+                                const uint32_t dilationW,
+                                const bool throwOnError) const {
+    auto error = kDwscLimits.kInputHWLimit.GetErrorOrEmpty(inHeight, inWidth);
+    const IsEqualToLimit kernels_count{inChannels, "number of kernels (must be equal to input channels count)"};
+    error += kernels_count.GetErrorOrEmpty(kernelN);
+    auto& inputChannelsNumberLimit = kDwscLimits.kInputChannelsNumberLimit;
```

Comment:
```suggestion
    const auto& inputChannelsNumberLimit = kDwscLimits.kInputChannelsNumberLimit;
```

Reply:
not agreed. deduction from assignment above will bring const type already, so extra const is not needed here. please re-set to resolved after reading.